src/api/app/models/bs_request_action.rb

```diff
@@ -482,20 +480,20 @@ def create_expand_package(packages, opts = {})
             raise BuildNotFinished, "The repository '#{pkg.project.name}' / '#{repo}' / #{arch} " \
                                     'needs recalculation by the schedulers'
           end
-          if result.attributes['state'].in?(['finished', 'publishing'])
+          if result.attributes['state'].value.in?(['finished', 'publishing'])
             raise BuildNotFinished, "The repository '#{pkg.project.name}' / '#{repo}' / #{arch}" \
                                     'did not finish the publish yet'
           end
-          unless result.attributes['state'].in?(['published', 'unpublished'])
+          unless result.attributes['state'].value.in?(['published', 'unpublished'])
             raise BuildNotFinished, "The repository '#{pkg.project.name}' / '#{repo}' / #{arch} " \
                                     'did not finish the build yet'
           end
 
           # all versrel are the same
           versrel[repo] ||= {}
-          result.get_elements('status').each do |status|
+          result.search('status').each do |status|
             package = status.attributes['package']
-            vrel = status.attributes['versrel']
+            vrel = status.attributes['versrel'].value if status.attributes['versrel']
```

Comment:
No point in having the assignment conditional if you `next` in the next line if it didn't happen